components/common/src/command/package/install.rs

```diff
@@ -63,7 +64,7 @@ use crate::{error::{Error,
             ui::{Status,
                  UIWriter}};
 
-pub const RETRIES: u64 = 5;
+pub const RETRIES: usize = 5;
 pub const RETRY_WAIT: u64 = 3000;
```

Comment:
As long as we're making modifications around here, I abhor constants without unit indications
```suggestion
pub const RETRY_WAIT_MS: u64 = 3000;
```
or now that `Duration` constructors are const functions, we could do even better:
```suggestion
pub const RETRY_WAIT: Duration = Duration::from_millis(3000);
```
but because the crate's implementation of `Fixed` only provides a `from_millis(u64)` constructor and `Duration`'s `as_millis` returns `u128`, we'd need a little helper function:
```rust
fn wait_for(delay: Duration, times: usize) -> impl IntoIterator<Item = Duration> {
    vec![delay].into_iter().cycle().take(times)
}
```
But I think this makes for the most ergonomic result:
```rust
retry(wait_for(RETRY_WAIT, RETRIES), …)
```

Reply:
It was such a small thing, I submitted [a PR to the `retry` crate](https://github.com/jimmycuadra/retry/pull/21).

Reply:
I like this suggestion, but it feels like `command/package/install.rs` is the wrong place for the `wait_for` function. It would be also used in key upload/download and Builder would also want to take advantage of it.  Somewhere in the core crate seems like the best place for it but I don't see an obvious (to me) spot for it.  What do you think?  

Would it be best to file an issue to change the type to a Duration once the retry crate accepts your pr?

Reply:
Yeah, I wouldn't put `wait_for` in install.rs. I'd suggest https://github.com/habitat-sh/habitat/blob/master/components/core/src/util/mod.rs as a good home that allows `builder` to take advantage.

No need to file an issue, I don't think. In the event that my PR is accepted into the `retry` crate, I'll get a notification and make the appropriate change once it merges. If you like, you could add a link to the PR in a doc comment for `wait_for` as a reminder.

Reply:
@baumanj I think I've addressed your feedback,  please let me know if there are any additional changes needed.

Reply:
Looks great! Thanks for your thoroughness